### PR TITLE
fix(auth): Added documentation for how to handle notauthorized exceptions when fail lambda is true

### PR DESCRIPTION
**Pull request by gpanshu**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [x] iOS
- [x] Android
- [x] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/auth/flutter/signin_with_custom_flow/40_custom_challenge.mdx [hunk 1]
```diff
@@ -13,3 +13,12 @@ Future<void> confirmSignIn(String generatedNumber) async {
 ```
 
 Once the user provides the correct response, they should be authenticated in your application.
+
+<Callout warning>
+
+<b>Special Handling on ConfirmSignIn</b>
+
+During a confirmSignIn call, if failAuthentication: true is returned by the Lambda, the session of the request gets invalidated by Cognito, and a NotAuthorizedException is thrown. To recover, the user must initiate a new sign in by calling Amplify.Auth.signIn.
```
Comment (dnys1) [inline src/fragments/lib/auth/flutter/signin_with_custom_flow/40_custom_challenge.mdx:21]:
```suggestion
During a `confirmSignIn` call, if `failAuthentication: true` is returned by the Lambda, the session of the request gets invalidated by Cognito, and a `NotAuthorizedException` is thrown. To recover, the user must initiate a new sign in by calling `Amplify.Auth.signIn`.
```

File: src/fragments/lib/auth/flutter/signin_with_custom_flow/40_custom_challenge.mdx [hunk 2]
```diff
@@ -13,3 +13,12 @@ Future<void> confirmSignIn(String generatedNumber) async {
...
+
+Exception: NotAuthorizedException with a message Invalid session for the user.
```
Comment (dnys1) [inline src/fragments/lib/auth/flutter/signin_with_custom_flow/40_custom_challenge.mdx:23]:
```suggestion
Exception: `NotAuthorizedException` with a message `Invalid session for the user.`
```

**Review by dnys1 — APPROVED** [review]

**Review by dnys1 — APPROVED** [review]

**Review by royjit — APPROVED** [review]

File: src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx [hunk 3]
```diff
@@ -94,6 +94,14 @@ RxAmplify.Auth.confirmSignIn(
...
 </Block>
+</BlockSwitcher>
 
+<Callout warning>
 
-</BlockSwitcher>
+<b>Special Handling on ConfirmSignIn</b>
+
+During a confirmSignIn call if `failAuthentication=true` is returned by the lambda the session of the request gets invalidated by cognito, a NotAuthorizedException is returned and a new signIn call is expected via Amplify.Auth.signIn
+```java
```
Comment (abdallahshaban557) [inline src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx:104]: do you need a Kotlin variant here for this code snippet?

**Comment by abdallahshaban557** [comment]
@gpanshu  - did you coordinate with the JS team making a change to their docs as well? Maybe check with @cwomack on this.

File: src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx (hunk 3 above)
Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx:104]: No this is just the exception and is kotlin/java agnostic.

**Comment by gpanshu** [comment]
> @gpanshu - did you coordinate with the JS team making a change to their docs as well? Maybe check with @cwomack on this.

I did and they do not have a page for custom auth but the PR has been shared with them and they will incorporate it when they create those docs.

File: src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx (hunk 3 above)
Comment (abdallahshaban557) [reply src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx:104]: Got it.

**Comment by abdallahshaban557** [comment]
Sounds good! Thank you @gpanshu !

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Review by div5yesh — APPROVED** [review]

**Review by tjleing — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**